File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.py
def hanoi(cnt, start, mid, end):
    if cnt == 1:
        print(f"{start} {end}")
        return

    hanoi(cnt-1, start, end, mid)
    print(f"{start} {end}")
    hanoi(cnt-1, mid, start, end)

n = int(input())
print(2**n-1)
hanoi(n, 1, 2, 3)

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	
	static StringBuilder sb = new StringBuilder();
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		sb.append((int)Math.pow(2, n)-1+"\n");
		hanoi(n, 1, 2, 3);
		
		System.out.println(sb);

	}
	
	public static void hanoi(int n, int start, int mid, int end) {
		if(n==1) {
			sb.append(start+" "+end+"\n");
			return;
		}
		
		hanoi(n-1, start, end, mid);
		sb.append(start+" "+end+"\n");
		hanoi(n-1, mid, start, end);
		
	}

}
